### ISSUE TITLE
Option to alias a different Fediverse instance

### ISSUE THREAD

**Issue by stevenjaycohen** [pfefferle/wordpress-activitypub#278]
If my prefered way  of interacting is through Pixelfed or Mastodon, I would love to be able to alias my WordPress blog so that people get that information from my profile.

Is this more complex than adding the URL of a different instance to the webfinger file? (Seriously, I am a bit confused on this front). Does webfinger use the term Alias in a different way than what I am understanding?

This should also lay the groundwork for eventually being able to migrate my other Fediverse accounts to all migrate to my WordPress blog (once the features are in place to interact from there).

**Comment by stevenjaycohen** [comment]
Code from my webfinger file (note: used the "hack" of writing the file manually)
[https://stevenjaycohen.com/.well-known/webfinger?resource=acct:sjc@stevenjaycohen.com](url)
`{
  "subject": "sjc@stevenjaycohen.com",
    "aliases": [
      "https://stevenjaycohen.com/author/sjc/",
      "https://mastodon.sdf.org/users/sjc"
    ],
    "links": [
    {
      "rel": "self",
      "type": "application/activity+json",
      "href": "https://stevenjaycohen.com/author/sjc/"
    },
    {
      "rel": "http://webfinger.net/rel/profile-page",
      "type": "text/html",
      "href": "https://stevenjaycohen.com/author/sjc/"
    }
    ]
}`

**Comment by pfefferle** [comment]
The spec says:

> The "aliases" array is an array of zero or more URI strings that identify the same entity as the "subject" URI.

So aliases are only meant to show different identifiers for one user on one plattform, it is not meant to forward identities across different plattforms.

ActivityPub (the spec) sadly does not define this kind of identity portability, but Mastodon has a proprietary way to handle that: https://blog.joinmastodon.org/2019/06/how-to-migrate-from-one-server-to-another/ but this might not work for Pixelfed or other plattforms.

**Comment by stevenjaycohen** [comment]
Got it! I am being too Mastodon centric. Makes sense. And yes, the link you sent is what I was thinking of. In messing around a bit, I did get both my blog and my mastodon account to show up when searching for my blog. Here's hoping that ActivityPub eventually evolves an optional but official way to connect the multiple identities that people have across the the Fediverse.

**Comment by pfefferle** [comment]
It is possible if you always use your an identifier (WebFinger) on your own domain that references a mastodon account, like maybe using a static webfinger file (simply copy the Mastodon WebFinger JSON to your server under `.well-known/webfinger`). This allows you to change the server behind as you need/wish, starting with Mastodon and switching to WordPress later on, for example. The only issue you have, is the friends network, that will only live on the plattform your WebFinger Indentifier points to and that you have to port to your new server.

An example:

The WebFinger Endpoint: https://stevenjaycohen.com/.well-known/webfinger?resource=acct:sjc@stevenjaycohen.com

Could return:

```JSON
{
  "subject": "acct:sjc@stevenjaycohen.com",
  "aliases": [
    "https://mastodon.sdf.org/@sjc",
    "https://mastodon.sdf.org/users/sjc",
    "https://stevenjaycohen.com/author/sjc/"
  ],
  "links": [
    {
      "rel": "http://webfinger.net/rel/profile-page",
      "type": "text/html",
      "href": "https://mastodon.sdf.org/@sjc"
    },
    {
      "rel": "self",
      "type": "application/activity+json",
      "href": "https://mastodon.sdf.org/users/sjc"
    },
    {
      "rel": "http://ostatus.org/schema/1.0/subscribe",
      "template": "https://mastodon.sdf.org/authorize_interaction?uri={uri}"
    }
  ]
}
```

**Comment by stevenjaycohen** [comment]
Yep, I played with that. But, I have bought into connecting my WordPress site to the Fediverse, going full Indie Web, and helping where I can by testing plugins like yours as they evolve :)

**Comment by stevenjaycohen** [comment]
> > Yep, I played with that. But, I have bought into connecting my WordPress site to the Fediverse, going full Indie Web, and helping where I can by testing plugins like yours as they evolve :)
> 
> So do you now have a wordpress blog connected to the fediverse AND some way to properly interact with the fediverse (like following others, replying) via Mastodon (or Pleroma, Friendica, etc) while keeping the same handle? Or do you use mastodon.sdf.org for this?

Until I can do everything from wordpress, I am using @sjc@mastodon.sdf.org to interact. My eventual hope is to be able to use mastodon's migrate function to eventually migrate my followers to my wordpress instance once it is full featured.
